src/types/fixedtimezone.jl

```diff
@@ -30,7 +30,7 @@ const FIXED_TIME_ZONE_REGEX = r"""
 A `TimeZone` with a constant offset for all of time.
 """
 struct FixedTimeZone <: TimeZone
-    name::String
+    name::Name
```

Comment:
Names here are quite different and we could probably restrict this  further to save on memory

Reply:
Can we restrict them to 15 characters outright?